### PR TITLE
[Firebase 11] Bump deps on GULs to v8 and GDT to v10

### PR DESCRIPTION
**Pull request by ncooke3**

### CODE REVIEW

**Comment by paulb777** [comment]
Will need to coordinate with the GDT major update that @andrewheard has started

**Comment by andrewheard** [comment]
> Will need to coordinate with the GDT major update that @andrewheard has started

@ncooke3 GDT 10.0 is [pushed](https://github.com/firebase/SpecsStaging/tree/main/GoogleDataTransport/10.0.0) to SpecsStaging now.

File: FirebaseInAppMessaging.podspec [hunk 1]
```diff
@@ -83,7 +83,7 @@ See more product details at https://firebase.google.com/products/in-app-messagin
   s.dependency 'FirebaseCore', '~> 10.0'
   s.dependency 'FirebaseInstallations', '~> 10.0'
   s.dependency 'FirebaseABTesting', '~> 10.0'
-  s.dependency 'GoogleUtilities/Environment', '~> 7.13'
+  s.dependency 'GoogleUtilities/Environment', '~> 8.0'
   s.dependency 'GoogleUtilities/UserDefaults', '~> 7.13'
```
Comment (andrewheard) [inline FirebaseInAppMessaging.podspec:87]: Please update `GoogleUtilities/UserDefaults` to `8.0` throughout.

File: FirebaseCrashlytics.podspec [hunk 2]
```diff
@@ -65,7 +65,7 @@ Pod::Spec.new do |s|
   s.dependency 'FirebaseRemoteConfigInterop', '~> 10.23'
   s.dependency 'PromisesObjC', '~> 2.1'
   s.dependency 'GoogleDataTransport', '~> 9.2'
```
Comment (andrewheard) [inline FirebaseCrashlytics.podspec:67]: Also `GoogleDataTransport` to `10.0` throughout.

**Comment by ncooke3** [comment]
CI still has many failures. I think the issue is that pods are being sourced from SpecsStaging and causing a mismatch in the GULs/GDT version between a local pod being tested and the staged pods it depends on. 

I think we may need to run the release tooling on this branch to update all the pod versions to 11.0.0 and stage those. WDYT?

**Comment by andrewheard** [comment]
> I think we may need to run the release tooling on this branch to update all the pod versions to 11.0.0 and stage those. WDYT?

I think that should work but you might need to comment out this line to prevent it from branching off `main`: https://github.com/firebase/firebase-ios-sdk/blob/02d963ce64a2f2187a058a3746e9ea8fa4ace986/ReleaseTooling/Sources/FirebaseReleaser/InitializeRelease.swift#L38

**Comment by paulb777** [comment]
I'd be ok to merge here and work through sample Podfile and quickstart issues separately.

**Review by paulb777 — APPROVED** [review]

**Pull request merged**